src/components/PaginatedVirtualList/utils.ts

```diff
@@ -11,19 +10,35 @@ type calculatePageSizeOptions = {
  * @param options.maxPageSize - The maximum number of items to render per page
  * @param options.totalItemCount - The total number of items in the list
  * @param options.currentPage - The current page number
- * @param options.offset - The additional number of items to render beyond the maxPageSize
  * @returns The number of items to render on the current page
  */
 const calculatePageSize = (options: calculatePageSizeOptions) => {
-  const { maxPageSize, totalItemCount, currentPage, offset } = options;
+  const { maxPageSize, totalItemCount, currentPage } = options;
   // The first page always has the maxPageSize or totalItemCount, whichever is smaller
   if (currentPage === 0) {
     return Math.min(maxPageSize, totalItemCount);
   }
 
-  // The rest of the pages have the pageSize + the offset, or the remaining items, whichever is smaller
   const remainingItems = totalItemCount - maxPageSize * currentPage;
-  return Math.min(maxPageSize + offset, remainingItems);
+
+  return Math.min(maxPageSize, remainingItems);
+};
+
+type CalculateStartingIndexOptions = {
+  page: number;
+  paginationThreshold: number;
+  paginationOffset: number;
+};
+/**
+ * `calculateStartingIndex` is used to calculate the starting index of the items to render on the current page.
```

Comment:
I think row index may be the better descriptor since we use line number to represent the real index of the line in the raw log file not including collapsed rows.